### ISSUE TITLE
Identityd Config Documentation

### ISSUE THREAD

**Issue by nhgs64** [Azure/iot-identity-service#67]
1. The [documentation for writing the identityd config file ](https://github.com/Azure/iot-identity-service/blob/main/docs/dev/running/aziot-identityd.md#configuring-and-running-aziot-identityd)seems to have some misleading parts:

>    "registration_id" = "<ADD DPS REGISTRATION ID HERE>" # Optional for X.509 attestation

The comment suggests this field is optional if using X.509 but in my experience is is required and must match the common name of the certificate?

```
><6>2020-11-06T19:48:42Z [INFO] - Starting service...
><6>2020-11-06T19:48:42Z [INFO] - Version - dev build
><3>2020-11-06T19:48:45Z [ERR!] - could not read config
><3>2020-11-06T19:48:45Z [ERR!] - caused by: missing field `registration_id` for key `provisioning`
```
2. Additionally, the service won't start without also including an undocumented field "global_endpoint":
```
<6>2020-11-06T19:48:42Z [INFO] - Starting service...
<6>2020-11-06T19:48:42Z [INFO] - Version - dev build
<3>2020-11-06T19:48:45Z [ERR!] - could not read config
<3>2020-11-06T19:48:45Z [ERR!] - caused by: missing field `registration_id` for key `provisioning`
```
Adding it in solved the problem but you might want to document this or remove the requirement:
>global_endpoint = "https://global.azure-devices-provisioning.net"

**Comment by arsing** [comment]
>>"registration_id" = "" # Optional for X.509 attestation
>
>The comment suggests this field is optional if using X.509 but in my experience is is required and must match the common name of the certificate?

It's supposed to be optional because the cert's CN can be used as the registration ID, just like it was with iotedged. It's a bug that the code requires it to be set.

---

>1. Additionally, the service won't start without also including an undocumented field "global_endpoint":
>
>Adding it in solved the problem but you might want to document this or remove the requirement:

That field ought to be documented and continue to be required, yes. (It's required because it has different values for other clouds.)

**Comment by arsing** [comment]
(By the way, you can use `aziot init` rather than generate the configs by hand; it'll do the right thing by default.)

**Comment by arsing** [comment]
We're planning to make the registration ID required instead, as mentioned in https://github.com/Azure/iot-identity-service/pull/123#issuecomment-786832599 . cc @daprilik

**Comment by daprilik** [comment]
Closed now that #146 is merged.
